runtime/src/error.c

```diff
@@ -125,26 +149,142 @@ static void chpl_stack_unwind(void){
     for(int t = 0; t < chpl_sizeSymTable; t+=2 ){
       if (!strcmp(chpl_funSymTable[t], buffer)){
 #ifdef __linux__
+        unw_proc_info_t info;
         // Maybe we can get a more precise line number
         unw_get_proc_info(&cursor, &info);
-        line = chpl_unwind_getLineNum(process_name,
-                                      (void *)(info.start_ip + wordValue));
+        line = chpl_unwind_getLineNum((void *)(info.start_ip + wordValue));
         // We wasn't able to obtain the line number, let's use the procedure
         // line number
         if(line == 0)
           line = chpl_filenumSymTable[t+1];
 #else
         line = chpl_filenumSymTable[t+1];
 #endif
-        fprintf(stderr,"%s() at %s:%d\n",
+        fprintf(out,"%s() at %s:%d%c",
                   chpl_funSymTable[t+1],
                   chpl_lookupFilename(chpl_filenumSymTable[t]),
-                  line);
+                  line,
+                  sep);
+        break;
+      }
+    }
+  }
+}
+
+// bufsz is the allocate size of the buffer
+// strsz is the number of bytes in the buffer currently used
+// str is the buffer
+// append is a 0-terminated string to append
+static void append_to_string(size_t* bufszArg, size_t* strszArg, char** strArg,
+                             const char* append) {
+  size_t toadd = strlen(append);
+  size_t bufsz = *bufszArg;
+  size_t strsz = *strszArg;
+  char* str = *strArg;
+
+  // allocate/reallocate the buffer if necessary
+  if (str == NULL) {
+    bufsz = 128 + toadd;
+    str = chpl_mem_alloc(bufsz, CHPL_RT_MD_IO_BUFFER, __LINE__, 0);
```

Comment:
Not sure, but I wonder if this should allocate from the system instead of our memory interface. If we're in the middle of a stack trace, can we go through our allocator (which will potentially require communication with other nodes)

Reply:
I'm not opposed to using the system allocator... but I don't see why we can't use our allocator? I don't think there is communication on allocation currently, right? Even if there were, it doesn't seem to prevent stack tracing. The stack trace is already saved; we are just converting it to text here.

Reply:
Our allocator can result in communication (e.g. under ugni if jemalloc needs more space, we will grab hugepages, register, and then broadcast registration addresses to all other nodes.)

I'm not sure I would expect that to prevent stack tracing, but I was thinking if we're printing a stack trace during a fatal error we wouldn't be able to rely on that.

Reply:
Right, but printing a stack trace during a fatal error won't run this code (it uses the other code that outputs to stdout/stderr).

Reply:
Ok, sounds good